### PR TITLE
For #11725: Switch to using WindowInsetsCompat instead of WindowInsets.

### PR DESCRIPTION
**Pull request by mcarare**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/support/utils/src/test/java/mozilla/components/support/utils/ext/WindowInsetsCompatTest.kt [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        whenever(windowInsetsCompat.getInsets(WindowInsetsCompat.Type.mandatorySystemGestures())).thenReturn(
+            mandatorySystemGestureInsets
+        )
+    }
+
+    @Test
+    fun testTop() {
+        val topInsets = windowInsetsCompat.top()
+
+        Assert.assertEquals(topPixels, topInsets)
```
Comment (csadilek) [inline components/support/utils/src/test/java/mozilla/components/support/utils/ext/WindowInsetsCompatTest.kt:51]: nit: we usually do a static import  of `Assert`.

**Review by csadilek — APPROVED** [review]
This all looks good to me. However, I think we should do some quick manual testing of the affected code paths, and land this next Tues. after we cut the branch. Maybe I am too conservative in this case, but I got burned by switching to a compat class before. :)

Arturo, wdyt?

File: components/support/ktx/src/main/java/mozilla/components/support/ktx/android/view/Activity.kt [hunk 2]
```diff
@@ -72,12 +63,7 @@ fun Activity.exitImmersiveModeIfNeeded() {
         return
     }
 
-    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
-        window.decorView.setOnApplyWindowInsetsListener(null)
-    } else {
-        @Suppress("DEPRECATION")
-        window.decorView.setOnSystemUiVisibilityChangeListener(null)
-    }
+    ViewCompat.setOnApplyWindowInsetsListener(window.decorView, null)
```
Comment (Amejia481) [inline components/support/ktx/src/main/java/mozilla/components/support/ktx/android/view/Activity.kt:66]: 🎉

**Review by Amejia481 — COMMENTED** [review]
This looks good to me too!
Thanks @mcarare for putting this patch, I agree with @csadilek as the fix doesn't have a high priority, we could wait until the next nightly cycle starts for landing.

**Comment by mcarare** [comment]
> I have already manually tested these changes, albeit only on the Sample Browser. I agree that we should wait for the cut to merge. Before that, I will also test all functionality in Fenix with local AC.

Tested Fenix with local AC on 2 devices. No regressions were observed. As the cut was made yesterday I will land this.

**Pull request merged**